File: benchmarks/cross-language/matrix_multiply/bench.py
def matrix_multiply(a: list[list[int]], b: list[list[int]]) -> list[list[int]]:
    rows_a = len(a)
    cols_a = len(a[0])
    cols_b = len(b[0])
    result = [[0] * cols_b for _ in range(rows_a)]
    for i in range(rows_a):
        for j in range(cols_b):
            for k in range(cols_a):
                result[i][j] += a[i][k] * b[k][j]
    return result

def main():
    size = 100
    a = [[(i + j) % 7 for j in range(size)] for i in range(size)]
    b = [[(i * j + 1) % 11 for j in range(size)] for i in range(size)]

    for _ in range(10):
        result = matrix_multiply(a, b)
    print(f"done: {result[0][0]}")

if __name__ == "__main__":
    main()

File: benchmarks/cross-language/matrix_multiply/bench.cs
using System;

class Program
{
    static long[][] MatrixMultiply(long[][] a, long[][] b)
    {
        int rowsA = a.Length;
        int colsA = a[0].Length;
        int colsB = b[0].Length;
        var result = new long[rowsA][];
        for (int i = 0; i < rowsA; i++)
        {
            result[i] = new long[colsB];
            for (int j = 0; j < colsB; j++)
            {
                long sum = 0;
                for (int k = 0; k < colsA; k++)
                    sum += a[i][k] * b[k][j];
                result[i][j] = sum;
            }
        }
        return result;
    }

    static void Main()
    {
        int size = 100;
        var a = new long[size][];
        var b = new long[size][];
        for (int i = 0; i < size; i++)
        {
            a[i] = new long[size];
            b[i] = new long[size];
            for (int j = 0; j < size; j++)
            {
                a[i][j] = (i + j) % 7;
                b[i][j] = (i * j + 1) % 11;
            }
        }

        long[][] result = null;
        for (int n = 0; n < 10; n++)
        {
            result = MatrixMultiply(a, b);
        }
        Console.WriteLine($"done: {result[0][0]}");
    }
}
